File: ps/minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string S) {
        int i, res=0, size=0;
        for(i=0;i<S.size();i++) {
            if (size==0 && S[i] == ')') res++;
            else if (S[i] == ')') size--;
            else size++;
        }
        return size+res;
    }
};

File: ps/minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        count = 0
        penalty = 0
        for ch in s:
            if ch == '(':
                count += 1
            else:
                count -= 1

            if count < 0:
                penalty += 1
                count = 0

        return count + penalty